sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -52,7 +52,7 @@ object ExternalCatalogUtils {
       '\n', '\u000B', '\u000C', '\r', '\u000E', '\u000F', '\u0010', '\u0011', '\u0012', '\u0013',
       '\u0014', '\u0015', '\u0016', '\u0017', '\u0018', '\u0019', '\u001A', '\u001B', '\u001C',
       '\u001D', '\u001E', '\u001F', '"', '#', '%', '\'', '*', '/', ':', '=', '?', '\\', '\u007F',
-      '{', '[', ']', '^')
+      '{', '[', ']', '^', '.')
```

Comment:
What about the behavior of Apache Hive?

Reply:
For me, it seems that Hive has the identical code.

https://github.com/apache/hive/blob/9857c4e584384f7b0a49c34bc2bdf876c2ea1503/common/src/java/org/apache/hadoop/hive/common/FileUtils.java#L251

Reply:
I don't know what happens in Hive as I don't know how to test that code, I am happy to verify the changes there if someone can explain how to test these changes in Hive. 

My understanding from reading https://github.com/apache/hive/blob/9857c4e584384f7b0a49c34bc2bdf876c2ea1503/common/src/java/org/apache/hadoop/hive/common/FileUtils.java#L300 is that any escaped sequence will be transformed correctly regardless of `clist` - it is only used for writes.

Reply:
> There should also be no backward compatibility issues since ExternalCatalogUtils allows reading both escaped and unescaped values.

I guess reading Hive tables are fine but it might break for Hive to read Spark's tables written.

Shall we better just add a configuration to control this? Also, I think we should better leave a comment that this part is different from Hive's (since we have comments above "The following string escaping code is mainly copied from Hive (o.a.h.h.common.FileUtils).").

Reply:
Actually, it would also break the forward compatibility too (old Spark versions to read the tables written by new Spark versions), which we don't guarantee in general but would be great to have a switch to control that in any event.

Reply:
I thought the code would translate any escaped values, so it should not matter what characters are encoded. As long as decoding code is generic enough to handle all escaped values, older Spark versions should be able to read it. I will verify this. For now, there is an issue in partitioning that prevents the PR to pass the tests.

Reply:
That would be great! If it can read/write these cases, I think that addresses @dongjoon-hyun's concern, and the change should be safe to go.

Reply:
I updated the PR description as I decided to keep all of the changes in one PR. The patch does introduce backward incompatible changes when writing double values as partition values.